### PR TITLE
Gh 774 init aou terraform

### PR DESCRIPTION
**Pull request by Fab-T**
Here is a fresh PR ready for review.

Test : Terraform good up until validate. (could not run plan b/c we need to enable API)

Notes : 
Some elements need to be reviewed ( TODO )

### CODE REVIEW

File: deploy/aou/terraform/aou-network.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
+module "aou-network" {
+  # TODO : update the source of the module with version
+  source = "github.com/broadinstitute/terraform-shared.git//terraform-modules/vpc-network?ref=hf_update_vpc"
+
+  providers = {
+    google      = google.broad-aou
+    google-beta = google-beta.broad-aou
+  }
+
+  network_name   = "aou-network"
```
Comment (hjfbynara) [inline deploy/aou/terraform/aou-network.tf:10]: I would be consistent with network name with what we have been using in other google projects.  I believe gotc-dev it is "app-network"

File: deploy/aou/terraform/aou-network.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  google_project = var.project_name
+  subnets = {
+    "us-central1" = {
+      cidr              = "10.128.0.0/20"
+      private_ip_access = true,
+      vpc_logging       = false,
+    },
+    "us-west1" = {
+      cidr              = "10.138.0.0/20"
+      private_ip_access = false,
```
Comment (hjfbynara) [inline deploy/aou/terraform/aou-network.tf:20]: turn on private ip access on for all US regions.  we will be attempting to spread processing across all US regions

File: deploy/aou/terraform/aou-providers.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
+# Configure the Google Cloud provider
+# Every google project that resources are going to be created in will have its own
+#  google provider definition using the google project name as its alias so as to be
+#  clear
+# As it does not accept variables, you must change the name in here and in global_vars
+
```
Comment (hjfbynara) [inline deploy/aou/terraform/aou-providers.tf:7]: Add providers for all the google projects that were created.  there will be some TF work in each of them.

File: deploy/aou/terraform/aou-cromwell-dns.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
+
+module "aou-cromwell-auth-proxy" {
+
+  source = "github.com/broadinstitute/gotc-deploy.git//terraform/auth_proxy?ref=tf_auth-proxy-0.0.1"
+
+  providers = {
+    google      = google.broad-aou
+    google-beta = google-beta.broad-aou
+  }
+  auth_proxy_dns_name    = "cromwell-aou.gotc-prod.broadinstitute.org"
```
Comment (hjfbynara) [inline deploy/aou/terraform/aou-cromwell-dns.tf:10]: I mentioned this in a previous comment.  dns_name is not fully qualified.   dns_zone will be gotc-prod and dns_project is broad-gotc-prod.   We leverage an existing DNS zone name that is managed in the gotc-prod project for prod DNS entries.

File: deploy/aou/terraform/aou-in-out-bucket.tf [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Input bucket and ACLs
+module "aou-in-out-buckets" {
```
Comment (hjfbynara) [inline deploy/aou/terraform/aou-in-out-bucket.tf:2]: general comment - not sure I like the way this module is done. hard to put in text.

File: deploy/aou/terraform/aou-network.tf (hunk 1 above)
Comment (Fab-T) [reply deploy/aou/terraform/aou-network.tf:10]: oh sorry this is what you meant in the previous comment...

File: deploy/aou/terraform/aou-providers.tf (hunk 3 above)
Comment (Fab-T) [reply deploy/aou/terraform/aou-providers.tf:7]: Need to understand how we can declare providers as we said we will be hosting all aou projects inside the same folder. Let's discuss this.

**Comment by Fab-T** [comment]
Need a fresh one

**Pull request closed without merging**